packages/components/progress-bar/src/progress-bar.scss

```diff
@@ -0,0 +1,109 @@
+$states: active, success, warning, error;
+
+:host {
+  --_border-radius: var(--sl-border-radius-full);
+  --_label-font: var(--sl-text-input-field-label-label-md);
+  --_helper-font: var(--sl-text-input-field-label-hint-md);
+  --_helper-color: var(--sl-color-input-helper-text-default);
+  --_helper-color-error: var(--sl-color-progressbar-error-track);
+  --_height: var(--sl-size-progressbar);
+  --_gap: var(--sl-space-progressbar-vertical-gap);
+  --_label-gap: var(--sl-space-progressbar-horizontal-gap);
+  --_indeterminate-duration: 20s;
+  --_bar-background: var(--sl-color-progressbar-background);
+
+  @media (prefers-reduced-motion: no-preference) {
+    --_indeterminate-duration: 1.2s;
+  }
+
+  color: var(--_helper-color);
+  display: flex;
+  flex-direction: column;
+  font: var(--_helper-font);
+  gap: var(--_gap);
+}
+
+@each $state in $states {
+  :host([state='#{$state}']) {
+    --_progress-background: var(--sl-color-progressbar-#{$state}-track);
+    --_icon-fill: var(--sl-color-progressbar-#{$state}-track);
+  }
+}
+
+:host([indeterminate][state='active']) {
+  .progress {
+    animation: var(--_indeterminate-duration) ease 500ms infinite normal none running indeterminate;
+    position: relative;
+  }
+}
+
+:host([state='error']) {
+  ::slotted(*) {
+    color: var(--_helper-color-error);
+  }
+}
+
+.label {
+  display: flex;
+  font: var(--_label-font);
+  gap: var(--_label-gap);
+  justify-content: space-between;
+
+  sl-icon {
+    --sl-icon-fill-default: var(--_icon-fill);
+  }
+}
+
+.helper {
+  display: flex;
```

Comment:
The `.helper` is a `<span>`, but you're styling it as a block element. Better to make it a `<div>` then.